runtime/vm/ContinuationHelpers.cpp

```diff
@@ -197,9 +210,13 @@ yieldContinuation(J9VMThread *currentThread)
 	 *
 	 * must be maintained for weakly ordered CPUs, to unsure that once the continuation is again available for GC scan (on potentially remote CPUs), all CPUs see up-to-date stack .
 	 */
-	Assert_VM_true((uintptr_t)currentThread == continuation->state);
+	assertContinuationMounted(continuation, currentThread);
 	continuation->state = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
```

Comment:
this is probably the last high-level continuation helper left that deals with the state directly, so let's convert this into a helper, too:

resetContinationState(continuation);